Source/embeddings/Builder/EmbeddingClassBuilder.ts

```diff
@@ -104,7 +104,7 @@ export class EmbeddingClassBuilder<T> implements ICanBuildAndRegisterAnEmbedding
     }
 
 
-    private createCompareMethod(method: CompareDecoratedMethod): EmbeddingCompareCallback<any> {
+    private createCompareMethod(method: UpdateDecoratedMethod): EmbeddingUpdateCallback<any> {
```

Comment:
```suggestion
    private createUpdateMethod(method: UpdateDecoratedMethod): EmbeddingUpdateCallback<any> {
```

We should also rename the `createRemoveMethod` to `createDeleteMethod`